### PR TITLE
Add SimulateTransaction RPC endpoint

### LINKED ISSUE #8986: Wallets would like to know if a transaction is valid before it is signed and submitted.
#### Problem

Wallets need to know if a transaction is valid before it is signed and submitted.

#### Proposed Solution

Expose an RPC to the validator that can execute an unsigned transaction and verify that the instructions are valid assuming all the signatures are valid.

TW requests that this does a signature check as well.  So allow fully signed transactions to be verified by the node but not forwarded to the leader.

tag: @CriesofCarrots

### PR DESCRIPTION
**Pull request by mvines**
RPC users need a mechanism to easily check if a transaction is valid before submitting it. 

Fixes #8986

### CODE REVIEW

File: docs/src/apps/jsonrpc-api.md [hunk 1]
```diff
@@ -1112,12 +1113,36 @@ Creates new transaction
 
 ```bash
 // Request
-curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","id":1, "method":"sendTransaction", "params":["3gKEMTuxvm3DKEJc4UyiyoNz1sxwdVRW2pyDDXqaCvUjGApnsazGh2y4W92zuaSSdJhBbWLYAkZokBt4N5oW27R7zCVaLLpLxvATL2GgheEh9DmmDR1P9r1ZqirVXM2fF3z5cafmc4EtwWc1UErFdCWj1qYvy4bDGMLXRYLURxaKytEEqrxz6JXj8rUHhDpjTZeFxmC6iAW3hZr6cmaAzewQCQfiEv2HfydriwHDtN95u3Y1EF6SuXxcRqox2aTjGye2Ln9zFj4XbnAtjCmkZhR"]}' http://localhost:8899
+curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","id":1, "method":"sendTransaction", "params":["4hXTCkRzt9WyecNzV1XPgCDfGAZzQKNxLXgynz5QDuWWPSAZBZSHptvWRL3BjCvzUXRdKvHL2b7yGrRQcWyaqsaBCncVG7BFggS8w9snUts67BSh3EqKpXLUm5UMHfD7ZBe9GhARjbNQMLJ1QD3Spr6oMTBU6EhdB4RD8CP2xUxr2u3d6fos36PD98XS6oX8TQjLpsMwncs5DAMiD4nNnR8NBfyghGCWvCVifVwvA8B8TJxE1aiyiv2L429BCWfyzAme5sZW8rDb14NeCQHhZbtNqfXhcp2tAnaAT"]}' http://localhost:8899
```
Comment (CriesofCarrots) [inline docs/src/apps/jsonrpc-api.md:1116]: Doesn't matter too much, but curious why this change?

File: core/src/rpc.rs [hunk 2]
```diff
@@ -1362,6 +1370,64 @@ impl RpcSol for RpcSolImpl {
...
+            );
+            return Err(Error::invalid_request());
+        }
+        let transaction: Transaction = bincode::config()
+            .limit(PACKET_DATA_SIZE as u64)
+            .deserialize(&data)
+            .map_err(|err| {
+                info!("simulate_transaction: deserialize error: {:?}", err);
+                Error::invalid_request()
+            })?;
```
Comment (CriesofCarrots) [inline core/src/rpc.rs:1396]: Move this to a helper fn (like `verify_pubkey`), since it is duplicated in `send_transaction`? Will also help ensure that `simulate` is checking the same things as `send`

File: docs/src/apps/jsonrpc-api.md (hunk 1 above)
Comment (mvines) [reply docs/src/apps/jsonrpc-api.md:1116]: The original value didn't deserialize anymore, it was probably added a while ago and not updated when we last changed the Transaction format

Comment (CriesofCarrots) [reply docs/src/apps/jsonrpc-api.md:1116]: Ah, no doubt. Maybe update the signature return as well, then?

Comment (mvines) [reply docs/src/apps/jsonrpc-api.md:1116]: lol I can't.  I get a blockhash expired error when I submit that transaction again

Comment (CriesofCarrots) [reply docs/src/apps/jsonrpc-api.md:1116]: Here, it's this: `2id3YC2jK9G5Wo2phDx4gJVAew8DcY5NAojnVuao8rkxwPYPe8cSwE5GzhEgJA2y8fVjDEo6iR6ykBvDxrTQrtpb`

Comment (mvines) [reply docs/src/apps/jsonrpc-api.md:1116]: ✔️

File: core/src/rpc.rs (hunk 2 above)
Comment (mvines) [reply core/src/rpc.rs]: Done (1a3b8cc7e2583fbd0aba69d409c47aed14e3ce36)
